### PR TITLE
tests: add "quiet" wrapper function that only prints output on failure

### PR DESCRIPTION
**Pull request by chipaca**
... and use it in the project's `prepare`.

### CODE REVIEW

File: tests/lib/quiet.sh [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+    tf="$(tempfile)"
+    if "$@" >& "$tf"; then
+        rm -f -- "$tf"
+        return 0
+    else
+        echo "Q: $*" >&2
+        cat "$tf" >&2
+        rm -f -- "$tf"
+        return 1
```
Comment (pedronis) [inline tests/lib/quiet.sh]: should we try to capture and use the actual exit code of the command for more complete info ?

File: tests/lib/quiet.sh [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    # note this is a subshell (parens instead of braces around the function)
+    # so this set only affects this function itself
+    { set +x; } >&/dev/null
+
+    tf="$(tempfile)"
+    if "$@" >& "$tf"; then
+        rm -f -- "$tf"
+        return 0
+    else
+        echo "Q: $*" >&2
```
Comment (niemeyer) [inline tests/lib/quiet.sh]: "quiet $*" would make it more clear. Suspect "Q:" won't mean much to most readers.

**Comment by chipaca** [comment]
http://pastebin.ubuntu.com/23811094/ is the diff between the two prepares.

File: tests/lib/quiet.sh (hunk 2 above)
Comment (chipaca) [inline tests/lib/quiet.sh]: you're right.

File: tests/lib/quiet.sh (hunk 1 above)
Comment (chipaca) [inline tests/lib/quiet.sh]: good idea

**Pull request merged**